### PR TITLE
[MISC] Rebranding into Discourse Gatekeeper

### PR DESCRIPTION
**Pull request by deusebio**
### Overview

We should rebrand the project as "Discourse Gatekeeper" following the decision about the naming for this project. 

> :warning: **IMPORTANT** :warning: After merging this PR, we should probably also change the name of the repository into discourse-gatekeeper

### Rationale

Daniele is suggesting us to rebrand, as the name `upload-charm-docs` is really not quite reflecting the true aim of this tool
 
### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/constants.py
Comment (jdkandersson) [inline src/constants.py:1]: These are breaking changes, could we implement support for both tags and include a deprecation warning for a while?

File: src/repository.py [hunk 1]
```diff
@@ -37,24 +37,24 @@
...
+    "tag created by discourse-gatekeeper to mark the latest push to discourse, managed by "
+    "discourse-gatekeeper, changes or removal of this tag may lead to unexpected behaviour"
 )
 
 CONFIG_USER_SECTION_NAME = "user"
 CONFIG_USER_NAME = (CONFIG_USER_SECTION_NAME, "name")
 CONFIG_USER_EMAIL = (CONFIG_USER_SECTION_NAME, "email")
 
-BRANCH_PREFIX = "upload-charm-docs"
+BRANCH_PREFIX = "discourse-gatekeeper"
```
Comment (jdkandersson) [inline src/repository.py:57]: This is also a breaking change for existing migration PRs, should we also support both for a while?

**Comment by deusebio** [comment]
Uhm, I'm not sure how breaking these changes are and how the deprecation should play out (if done within the same repo, see below). I mean, these constants are not provided by the user, but are used internally, and exposed in the form of tag names, branch names, pull-requests infos. 
Note that if the tag with the name does not exists, it gets created anyway, fixing the repo itself. Given this, I'm afraid that creating extra code to handle both naming in the same repository would only increase the engineering burder

The biggest change is actually probably changing the name of the repository (since it then requires people to update their actions), but there is no deprecation we can do about this. 

One way to go could be to fork this project into a new repository with the right name (discourse-gatekeeper) and merge this PR there. In this repository we place a logging saying that this action is now deprecated (and not supported), nudging people to use the other one that will be the supported one. This should also allow us to have cleaner code

**Comment by jdkandersson** [comment]
> Uhm, I'm not sure how breaking these changes are and how the deprecation should play out (if done within the same repo, see below). I mean, these constants are not provided by the user, but are used internally, and exposed in the form of tag names, branch names, pull-requests infos. Note that if the tag with the name does not exists, it gets created anyway, fixing the repo itself. Given this, I'm afraid that creating extra code to handle both naming in the same repository would only increase the engineering burder
> 
> The biggest change is actually probably changing the name of the repository (since it then requires people to update their actions), but there is no deprecation we can do about this.
> 
> One way to go could be to fork this project into a new repository with the right name (discourse-gatekeeper) and merge this PR there. In this repository we place a logging saying that this action is now deprecated (and not supported), nudging people to use the other one that will be the supported one. This should also allow us to have cleaner code

You are right actually, if the tag gets re-created that is fine. I think that GitHub maintains redirects so the renaming shouldn't be an issue

**Review by jdkandersson — DISMISSED** [review]

**Comment by deusebio** [comment]
@jdkandersson should we merge this PR? I have merged latest update of `main`, but I believe one reviewer is still required to have a look and approve the PR

**Review by jdkandersson — APPROVED** [review]

**Comment by jdkandersson** [comment]
> @jdkandersson should we merge this PR? I have merged latest update of `main`, but I believe one reviewer is still required to have a look and approve the PR

I'll ask for a review from the team

File: src/__init__.py [hunk 2]
```diff
@@ -25,8 +25,8 @@
 )
 
 GETTING_STARTED = (
-    "To get started with upload-charm-docs, "
-    "please refer to https://github.com/canonical/upload-charm-docs#getting-started"
+    "To get started with discourse-gatekeeper, "
+    "please refer to https://github.com/canonical/discourse-gatekeeper#getting-started"
```
Comment (yanksyoon) [inline src/__init__.py:29]: Out of curiosity, will the repository be renamed as well?

File: src/constants.py [hunk 3]
```diff
@@ -7,8 +7,8 @@
 another module or to resolve circular imports.
 """
 DEFAULT_BRANCH = "main"
-DOCUMENTATION_TAG = "upload-charm-docs/base-content"
-DISCOURSE_AHEAD_TAG = "upload-charm-docs/discourse-ahead-ok"
+DOCUMENTATION_TAG = "discourse-gatekeeper/base-content"
+DISCOURSE_AHEAD_TAG = "discourse-gatekeeper/discourse-ahead-ok"
```
Comment (yanksyoon) [inline src/constants.py:11]: I believe this would be a breaking change, probably something to note for repositories already using it!

**Review by yanksyoon — APPROVED** [review]
LGTM, minor comments thank you!

File: src/__init__.py (hunk 2 above)
Comment (deusebio) [reply src/__init__.py:29]: Yes, it should. @jdkandersson was saying [here](https://github.com/canonical/upload-charm-docs/pull/210#issuecomment-1710229308) that redirects with the old name would still work

File: src/constants.py (hunk 3 above)
Comment (deusebio) [reply src/constants.py:11]: Not really, see [my comment above](https://github.com/canonical/upload-charm-docs/pull/210#issuecomment-1709627616)

**Comment by deusebio** [comment]
I had to fix some conflicts on `src-docs`, which dismissed reviews. 

Please reapprove @jdkandersson  and @yanksyoon (BTW, I hope I also replied to your minors)

**Comment by deusebio** [comment]
@yanksyoon could you re-approve? I had to rebase the branch, but that cleared reviews :(

**Comment by jdkandersson** [comment]
> @yanksyoon could you re-approve? I had to rebase the branch, but that cleared reviews :(

He's been away unwell, I'll ask for review in our team

**Review by cbartz — APPROVED** [review]
LGTM

**Comment by deusebio** [comment]
> He's been away unwell, I'll ask for review in our team

Oh, I see :( . I hope he will recover and get better soon! @yanksyoon

**Pull request merged**